packages/xo-web/src/xo-app/dashboard/health/index.js

```diff
@@ -510,12 +510,15 @@ const HANDLED_VDI_TYPES = new Set(['system', 'user', 'ephemeral'])
   const getVifsByMac = createGetObjectsOfType('VIF')
     .pick(
       createCollectionWrapper(
-        createSelector(getVms, vms =>
-          flatMap(
-            filter(vms, ({ blockedOperations: ops }) => !('start' in ops || 'start_on' in ops)),
-            'VIFs'
-          ).sort()
-        )
+        createSelector(getVms, vms => {
+          let vifs = []
+          forEach(vms, ({ blockedOperations: ops, VIFs }) => {
+            if (!('start' in ops || 'start_on' in ops)) {
+              vifs = [...vifs, ...VIFs]
```

Comment:
IMHO, this much worst than a filter+map, it looks like a `O(n²)`.

Reply:
This change is outdated. Do you agree with [this](https://github.com/vatesfr/xen-orchestra/pull/5916/files#diff-2af4feb739063d24cfb951db648985938b2292a9e5edc862aec3065b6d8a74faR517) though?

Reply:
Oh! My bad, yes, your change is good :slightly_smiling_face: